ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Util/LeakyBucket.hs

```diff
@@ -367,7 +368,8 @@ leak runThreadVar actionThreadId bucket = go
           go
         else do
           -- Wait for the bucket to empty, or for the thread to be stopped or
-          -- restarted.
+          -- restarted. Beware not to call 'registerDelay' with argument 0, that
+          -- is ensure that @timeToWait >= microsecondsPerSecond@.
           varTimeout <- registerDelay $ diffTimeToMicrosecondsAsInt timeToWait
```

Comment:
Hmm, why `registerDelay` (before 5abc38f56150d3a5cce7b98b30c650b44c74600b) ever be called with 0 as its argument? Wouldn't we be in the first case of the `if` above (as its condition contains `|| timeToWait == 0`)? Was it a rounding thing (in which case we should probably check `diffTimeToMicrosecondsAsInt timeToWait == 0`)?

Reply:
`DiffTime` has picosecond precision while `registerDelay` only has microsecond precision, so even when `timeToWait /= 0` there islots of room for `diffTimeToMicrosecondsAsInt timeToWait == 0`. I suppose we could check the latter indeed.

Reply:
>  which case we should probably check `diffTimeToMicrosecondsAsInt timeToWait == 0`

1970291542a1f4fa3ec4c4bf2604dcc3bccf4b45